tests/cpputest/testfunctions.cpp

```diff
@@ -10,75 +10,71 @@
 #include <cmath>
 #include <unistd.h>
 
+
+extern amici::Model *getModel(const amici::UserData *udata);
```

Comment:
See comment above. I would not use those extern functions anywhere but in top-level code. It makes it very hard so see the dependencies. I don't see a downside to passing the model object as additional argument. (Maybe except for the UserDataFromHDF5, where it does not seem very plausible but that can be fixed).